pkg/ignition/service_config.go

```diff
@@ -59,3 +59,37 @@ func (b *ignitionBuilder) authFile() ignition_config_types_32.File {
 		FileEmbedded1: ignition_config_types_32.FileEmbedded1{Contents: ignition_config_types_32.Resource{Source: &source}},
 	}
 }
+
+func (b *ignitionBuilder) hostNameFixService() ignition_config_types_32.Unit {
+	script := fmt.Sprintf(`
+if [ "$(hostnamectl hostname)" =~ ^localhost(\.localdomain)?$ ]]; then
```

Comment:
Since you aren't removing the [dispatcher script](https://github.com/openshift/image-customization-controller/blob/main/pkg/ignition/builder.go#L123) (which I think is a good thing), I don't think this will ever be true.

I think what this service wants to do is detect if we only have a --transient hostname set and convert it to a --static one.

If /etc/hostname gets copied by coreos-installer then that will work, at the cost of it being baked in such that even the later appearance of a DHCP hostname will not cause it to change (maybe that's a good thing?).

If we want this to happen as late in the process as possible, maybe IPA itself should do it just before calling coreos-installer?

Reply:
> If /etc/hostname gets copied by coreos-installer

https://github.com/coreos/fedora-coreos-tracker/issues/697 suggests that it doesn't, as recently as March.

Reply:
Okay, then we're back to the ignition approach, I'm afraid.